src/OrchardCore.Modules/OrchardCore.Media/Indexing/MediaFieldIndexHandler.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell.Builders;
+using OrchardCore.Indexing;
+using OrchardCore.Media.Fields;
+using OrchardCore.Media.Settings;
+
+namespace OrchardCore.Media.Indexing
+{
+    public class MediaFieldIndexHandler : ContentFieldIndexHandler<MediaField>
+    {
+        private const string MediaTextKeySuffix = ".MediaText";
+        private const string FileTextKeySuffix = ".FileText";
+
+        private readonly IMediaFileStore _mediaFileStore;
+        private readonly MediaFileIndexingOptions _mediaFileIndexingOptions;
+        private readonly IServiceProvider _serviceProvider;
+
+        public MediaFieldIndexHandler(
+            IMediaFileStore mediaFileStore,
+            IOptions<MediaFileIndexingOptions> mediaFileIndexingOptions,
+            IServiceProvider serviceProvider)
+        {
+            _mediaFileStore = mediaFileStore;
+            _mediaFileIndexingOptions = mediaFileIndexingOptions.Value;
+            _serviceProvider = serviceProvider;
+        }
+
+        public async override Task BuildIndexAsync(MediaField field, BuildFieldIndexContext context)
+        {
+            var options = context.Settings.ToOptions();
+            var settings = context.ContentPartFieldDefinition.GetSettings<MediaFieldSettings>();
+
+            if (field.Paths.Length > 0)
+            {
+                if (settings.AllowMediaText)
+                {
+                    foreach (var key in context.Keys)
+                    {
+                        foreach (var mediaText in field.MediaTexts)
+                        {
+                            context.DocumentIndex.Set(key + MediaTextKeySuffix, mediaText, options);
+                        }
+                    }
+                }
+
+                if (_mediaFileIndexingOptions.AnyMediaFileTextProviders())
```

Comment:
This check is actually unecesary, because this is not a feature, so pdf indexing is always enabled, so there will always be a registered pdf indexer.

Unless you implement a `Remove` method (I don't see that we need it, so I'd just drop the `.Any()` check)

Reply:
Indeed, though defensive checks like this are useful if the feature is changed later. Anyway, removed it.